lib/operations/collection_ops.js

```diff
@@ -55,6 +61,118 @@ const toError = require('../utils').toError;
 const groupFunction =
   'function () {\nvar c = db[ns].find(condition);\nvar map = new Map();\nvar reduce_function = reduce;\n\nwhile (c.hasNext()) {\nvar obj = c.next();\nvar key = {};\n\nfor (var i = 0, len = keys.length; i < len; ++i) {\nvar k = keys[i];\nkey[k] = obj[k];\n}\n\nvar aggObj = map.get(key);\n\nif (aggObj == null) {\nvar newObj = Object.extend({}, key);\naggObj = Object.extend(newObj, initial);\nmap.put(key, aggObj);\n}\n\nreduce_function(obj, aggObj);\n}\n\nreturn { "result": map.values() };\n}';
 
+/**
+ * Perform an aggregate operation. See Collection.prototype.aggregate for more information.
+ *
+ * @method
+ * @param {Collection} a Collection instance.
+ * @param {object} [pipeline=[]] Array containing all the aggregation framework commands for the execution.
+ * @param {object} [options] Optional settings. See Collection.prototype.aggregate for a list of options.
+ * @param {Collection~aggregationCallback} callback The command result callback
+ */
+const aggregateSchema = {
```

Comment:
is this just the same schema defined above in `collection`? Can you give a bit of detail here about the need to duplicate, and why we have `applyDefaults`?